tests/conftest.py

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from copy import deepcopy
+
+import pytest
+
+from airflow.utils import db
+
+
+@pytest.fixture(autouse=True)
+def reset_environment():
+    """
+    Resets env variables.
+    """
+    init_env = deepcopy(os.environ)
```

Comment:
```suggestion
    init_env =os.environ.copy()
```
In python standard library, we have a dedicated method for this.

Reply:
How about using monkeypatch ? https://docs.pytest.org/en/latest/monkeypatch.html#monkeypatching-environment-variables - I guess you wanted to be 100% sure that we will restore all env variables instead of changing all the places where os.environ gets modified?

I checked that there seem to be a handful of places where environ is modified. 
I know it is safer to save the env, but on the other hand it might be a lot slower:

<img width="965" alt="Screenshot 2019-11-25 at 22 39 53" src="https://user-images.githubusercontent.com/595491/69580300-9675b000-0fd4-11ea-8975-acead3f3cd7a.png">

WDYT?

Reply:
I agree. Instead of monkeypatch we can mark those test with `usefixture` and use the fixture in question, WDYT?

Reply:
I checked all those occurrences and fixed `test_hive_hook` in other places it's used in helper or handled properly.

Reply:
@nuclearpinguin ^^ monkeypatch seems nicer and pytest-dedicated way of temporary modifying the environment (and other things). It is - far less code and it restores environment automatically. So if we decide to optimise the speed and get-rid of auto-fixture, I think monkeypatch is better solution.